src/plugins/intel_gna/tests/unit/gna_memory_alignment.cpp

```diff
@@ -0,0 +1,164 @@
+            gnadevice.reset();
+        }
+    }
+
+    const size_t get_memory_REGION_INPUTS_size() const {
+        return this->gnamem->getQueue(ov::intel_gna::memory::REGION_INPUTS)->calcSize();
+    }
+};
+
+class GNAPluginLoadNetworkTests : public ::testing::TestWithParam<MemAlignmentTestParams> {
+public:
+    static std::string GetTestCaseName(const testing::TestParamInfo<MemAlignmentTestParams>& obj) {
+        ExecutionMode exe_mode;
+        HWGeneration hw_gen;
+        std::pair<ngraph::Shape, size_t> inp_shape_vs_mem;
+        tie(exe_mode, hw_gen, inp_shape_vs_mem) = obj.param;
+
+        std::ostringstream result;
+        result << "inp=" << inp_shape_vs_mem.first.to_string() << "_";
+        result << "mem_region_size=" << inp_shape_vs_mem.second;
+        return result.str();
+    }
+
+protected:
+    void Run() {
+        ExecutionMode exe_mode;
+        HWGeneration hw_gen;
+        std::pair<ngraph::Shape, size_t> inp_shape_vs_mem;
+        tie(exe_mode, hw_gen, inp_shape_vs_mem) = this->GetParam();
+        ngraph::Shape inp_shape = inp_shape_vs_mem.first;
+        size_t mem_region_size = inp_shape_vs_mem.second;
+
+        const ov::AnyMap gna_config = {ov::intel_gna::execution_mode(exe_mode), ov::intel_gna::compile_target(hw_gen)};
+
+        auto plugin = GNAPluginForMemoryAlignmentTest(any_copy(gna_config));
+        auto function = getMulFunction(inp_shape);
+        CNNNetwork cnnNetwork(function);
+        plugin.LoadNetwork(cnnNetwork);
+        EXPECT_EQ(plugin.get_memory_REGION_INPUTS_size(), mem_region_size);
+    }
+
+    void SetUp() override {
+        test_params = GetParam();
+    }
+
+private:
+    std::shared_ptr<ov::Model> getMulFunction(const ngraph::Shape input_shape) {
+        const ngraph::element::Type net_precision = ngraph::element::f32;
+
+        auto input = std::make_shared<ngraph::opset8::Parameter>(net_precision, input_shape);
+        auto multiplier = std::make_shared<ngraph::opset8::Constant>(net_precision, input_shape);
+        auto matmul = std::make_shared<ngraph::opset8::MatMul>(input, multiplier, false, true);
+        auto result = std::make_shared<ngraph::opset8::Result>(matmul);
+        auto function = std::make_shared<ov::Model>(ov::ResultVector({result}), ov::ParameterVector({input}), "MatMul");
+        return function;
+    }
+
+    MemAlignmentTestParams test_params;
+};
+
+TEST_P(GNAPluginLoadNetworkTests, CompareInpShapeVsReservedMemRegion) {
+    Run();
+}
+
+INSTANTIATE_TEST_SUITE_P(MemoryAlignment_FP32,
+                         GNAPluginLoadNetworkTests,
+                         ::testing::Combine(::testing::Values(ExecutionMode::SW_FP32),
+                                            ::testing::Values(HWGeneration::UNDEFINED),
+                                            ::testing::ValuesIn(param_16B_alignment_prec_fp32)),
+                         GNAPluginLoadNetworkTests::GetTestCaseName);
+
+INSTANTIATE_TEST_SUITE_P(MemoryAlignment_GNA_3_0,
```

Comment:
It's a cosmetic, but why cannot we combine these two test suites into one if the only difference is the hw generation parameter?